protocol/flows/src/flow_context.rs

```diff
@@ -128,6 +137,11 @@ impl FlowContext {
         self.is_ibd_running.load(Ordering::SeqCst)
     }
 
+    pub async fn is_nearly_synced(&self) -> bool {
+        // TODO: make sure relying on is_ibd_running() is safe in all cases
+        !self.is_ibd_running() || self.consensus().session().await.is_nearly_synced()
```

Comment:
I would remove `!self.is_ibd_running() ||`. Also I think callers should obtain the session themselves and make a direct call (in one of current usages a session is obtained anyway). So please remove it altogether